odbc/src/api.rs

```diff
@@ -104,7 +108,17 @@ pub extern "C" fn SQLBindParameter(
     _buffer_length: Len,
     _str_len_or_ind_ptr: *mut Len,
 ) -> SqlReturn {
-    unimplemented!()
+    match add_diag_info(
+        HandleType::Stmt,
+        hstmt as *mut _,
+        UNIMPLEMENTED_FUNC.to_string(),
+        "SQLBindParameter is unimplemented".to_string(),
+        0,
+        "api".to_string(),
+    ) {
+        Ok(_) => SqlReturn::ERROR,
+        Err(_) => SqlReturn::INVALID_HANDLE,
```

Comment:
instead of having every function do this mapping itself, would it instead make sense for `ODBCError` to have a `return_code` method that returns the appropriate return code for the error?

Reply:
I actually changed the return type of `add_diag_info`, so now we can just return `SqlReturn::ERROR` in lieu of a match statement. The [docs](https://docs.microsoft.com/en-us/sql/odbc/reference/the-driver-manager?view=sql-server-ver15) say that the driver manager will check for null pointers, so the driver shouldn't need to return `SqlReturn::INVALID_HANDLE`.